### PR TITLE
Only check type-stability of frule and rrule if primal is type-stable

### PR DESCRIPTION
**Pull request by sethaxen**
This PR follows up #78 to only test that `frule`s and `rrule`s are type-stable when the primal itself is type-stable.

### CODE REVIEW

**Comment by sethaxen** [comment]
@oxinabox I guess this could be included as part of v0.6.0, but I've incremented the version number anyways.

File: Project.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name = "ChainRulesTestUtils"
 uuid = "cdddcdb0-9152-4a09-a978-84456f9df70a"
-version = "0.6.0-DEV"
+version = "0.6.1"
```
Comment (oxinabox) [inline Project.toml:3]: Can't make a patch release til we do an actual release
but lets do an actual release with this
```suggestion
version = "0.6.0"
```

File: src/testers.jl [hunk 2]
```diff
@@ -188,6 +188,20 @@ function _test_inferred(f, args...; kwargs...)
     end
 end
 
+"""
+    _is_typestable(f, args...; kwargs...) -> Bool
+
+Return whether `f(args...; kwargs...)` is type-stable.
+"""
+function _is_typestable(f, args...; kwargs...)
```
Comment (oxinabox) [inline src/testers.jl:196]: Technically this is not checking type-stability, it is checking inferability.
Julia doesn't promise that all type-stable things will infer.

File: src/testers.jl [hunk 3]
```diff
@@ -197,7 +211,8 @@ end
 - `ẋ`: differential w.r.t. `x` (should generally be set randomly).
 
 `fkwargs` are passed to `f` as keyword arguments.
-If `check_inferred=true`, then the type-stability of the `frule` is checked.
+If `check_inferred=true`, then the type-stability of the `frule` is checked, as long as `f`
+is itself type-stable.
```
Comment (oxinabox) [inline src/testers.jl:215]: similar here (I should have caught this before).
Inferable implies type stable
type stable does not imply inferable.

**Review by oxinabox — APPROVED** [review]
LGTM baring a minor comment.
Merge once addressed

File: Project.toml (hunk 1 above)
Comment (oxinabox) [reply Project.toml:3]: Actually can keave it at 0.6.1 and I am releasing 0.6 presently

**Pull request merged**